File: python/arrays-and-hashing/remove-duplicates-from-sorted-array.py
from typing import List
def removeDuplicates(nums: List[int]) -> int:
        l = 1

        for r in range(1, len(nums)):
            # check previous value versus current
            if nums[r] != nums[r - 1]:
                # if not equal updates current value to the one that is ahead, effectively removing the duplicate
                nums[l] = nums[r]
                # updates l to count the length of the array which is what we want to return
                
                l += 1
                print(l)
        return l


removeDuplicates([1,1,2])
removeDuplicates([0,0,1,1,1,2,2,3,3,4])

File: javascript/arrays-and-hashing/remove-duplicates-from-sorted-array.js
var removeDuplicates = function(nums) {
    // return the array with duplicates removed

    // initialize the result counter
    let counter = 1;

    // loop to interate over the array
    for(let i = 1; i < nums.length; i++ ) {
        
        // checks the current number against the previous one
        if(nums[i] !== nums[i-1]) {

            // updates the current nums i position with counter position
            nums[counter] = nums[i];
            counter++;
            
        }
        
    }
    
    return counter;
}

console.log(removeDuplicates([1,1,2]))
console.log(removeDuplicates([0,0,1,1,1,2,2,3,3,4]))